pkg/probe/http/http_test.go

```diff
@@ -364,48 +373,60 @@ func TestHTTPProbeChecker(t *testing.T) {
 			accBody: "Probe terminated redirects, Response body: ok body",
 		},
 	}
-	for i, test := range testCases {
-		t.Run(fmt.Sprintf("case-%2d", i), func(t *testing.T) {
-			server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-				test.handler(w, r)
-			}))
-			defer server.Close()
-			u, err := url.Parse(server.URL)
-			if err != nil {
-				t.Errorf("case %d: unexpected error: %v", i, err)
-			}
-			_, port, err := net.SplitHostPort(u.Host)
-			if err != nil {
-				t.Errorf("case %d: unexpected error: %v", i, err)
-			}
-			_, err = strconv.Atoi(port)
-			if err != nil {
-				t.Errorf("case %d: unexpected error: %v", i, err)
-			}
-			req, err := NewProbeRequest(u, test.reqHeaders)
-			if err != nil {
-				t.Fatal(err)
-			}
-			health, output, err := prober.Probe(req, 1*time.Second)
-			if test.health == probe.Unknown && err == nil {
-				t.Errorf("case %d: expected error", i)
-			}
-			if test.health != probe.Unknown && err != nil {
-				t.Errorf("case %d: unexpected error: %v", i, err)
-			}
-			if health != test.health {
-				t.Errorf("case %d: expected %v, got %v", i, test.health, health)
-			}
-			if health != probe.Failure && test.health != probe.Failure {
-				if !strings.Contains(output, test.accBody) {
-					t.Errorf("Expected response body to contain %v, got %v", test.accBody, output)
+	for _, testHTTP := range tests {
+		for i, test := range testCases {
+			t.Run(fmt.Sprintf("%s-case-%2d", testHTTP.name, i), func(t *testing.T) {
+				server := httptest.NewUnstartedServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
```

Comment:
Can you comment on what, if anything, proves connection reuse?

Reply:
I'm not able to make it work, I will leave it for a follow up